broadinstitute/rawls
# QA-1974: Run Rawls tests by tag. [Good for review]

**Description**

JIRA ticket [QA-1974](https://broadworkbench.atlassian.net/browse/QA-1974). See also parent ticket [QA-1970](https://broadworkbench.atlassian.net/browse/QA-1970).

Group Rawls tests so they can be run by following tags (see [Rawls Swat Test Ownership](https://docs.google.com/spreadsheets/d/1YxghKs-16IRxJ6R6N9HoAKGc3I6ly2MTaCuMuAay1nE/edit#gid=0))

- AuthDomainsTest
- BillingsTest
- DataRepoSnapshotsTest
- MethodsTest
- RawlsTest
- WorkspacesTest

There is an associated PR in `dsp-jenkins` [#616](https://github.com/broadinstitute/dsp-jenkins/pull/616) that goes with this ticket.

The following new Jenkins job demonstrates how the router works for PRs.

- [https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline-rawls/](https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline-rawls/)
- [https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-fiab-test-router/](https://fc-jenkins.dsp-techops.broadinstitute.org/job/rawls-fiab-test-router/)

Slack notifications will be handled in another ticket.

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
